### PR TITLE
import more hoc lesson plans from curriculum builder

### PR DESCRIPTION
**Pull request by davidsbailey**
Continues https://codedotorg.atlassian.net/browse/PLAT-1265. See https://github.com/code-dot-org/code-dot-org/pull/42567 for background.

Mostly this PR contains lessons that were blocked from import by not having the CSTA2011 standards, which have since been added in #42580.

`text-compression` and `hoc-encryption` were also blocked because they depended on pre-2020 CSP standards, which hadn't been imported yet. per https://github.com/code-dot-org/code-dot-org/pull/42582, the decision was to strip these outdated standards for now (cb2adb7c8ee08c6bcd64b2b440222c5f5baad74e) so that we can keep the older CSP standards out of our system.

`hourofcode` is interesting because its script levels live on studio.code.org/hoc/1, not /s/hoc/1. however, the unit overview page is still located at /s/hourofcode, and I didn't have to do anything extra to get the import to work for this unit.

You can see a demo of this round of imported lessons at:
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/starwarsblocks 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/starwars 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/frozen 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/basketball 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/sports 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/flappy 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/infinity 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/playlab 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/iceage 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/gumball 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/artist 
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/hoc-encryption
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/text-compression
* https://adhoc-import-hoc-lesson-studio.cdn-code.org/s/hourofcode

adhoc login credentials [here](https://docs.google.com/document/d/1eQKbVbjqMLjj0KX6QodZE7NrzxDQBy50oVnkMK76J8Q/edit#heading=h.uuz48noudezc).

These are the commands I ran to do the import:
```
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u starwars,starwarsblocks -p /hoc/plugged/5 &&\
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u frozen -p /hoc/plugged/2 &&\
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u flappy -p /hoc/plugged/3

bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u infinity,playlab,iceage,gumball -p /hoc/plugged/4  

bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u basketball,sports -p /csf-20/coursed/5 &&\
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u artist -p /csf-20/coursec/7

bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u hoc-encryption -p /hoc/unplugged/3
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u text-compression -p /hoc/unplugged/2
bundle exec bin/curriculum/import_hoc_lesson_plan.rb -v -u hourofcode -p /hoc/plugged/1
```

## Testing story

I verified the following things locally:
* `rake seed:all` succeeds
* For each imported script, starting from the script overview page:
  * View Lesson Plan button goes to the right legacy lesson plan
  * gray links box takes you to the code studio lesson plan
  * code studio lesson plan page loads properly, and appears to have the right content

## Follow up work

There is one remaining hoc script, `outbreak`, to import. that one will require special handling because the lesson plan data we want to import is already in another lesson on code studio.

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/curriculum/text-compression/1/Teacher.moved
Comment (davidsbailey) [inline pegasus/sites.v3/code.org/public/curriculum/text-compression/1/Teacher.moved:1]: this file was different in that it existed already, but was pointing to the wrong location. however, the old lesson plan was not previously accessible, because has_lesson_plan was false, so the View Lesson Plan button was not visible.

**Comment by dmcavoy** [comment]
> `hourofcode` is interesting because its script levels live on studio.code.org/hoc/1, not /s/hoc/1.

Would it be good to get these to live on /s/hoc/1 and just have /hoc/ redirect to there if someone has that old link?

**Review by dmcavoy — APPROVED** [review]
This looks good to me. I tagged Mike in case he has any thoughts before we merge.

**Comment by davidsbailey** [comment]
> > `hourofcode` is interesting because its script levels live on studio.code.org/hoc/1, not /s/hoc/1.
> 
> Would it be good to get these to live on /s/hoc/1 and just have /hoc/ redirect to there if someone has that old link?

good question. ideally this would be on /s/hoc/1, but my first thought is that I would lean toward not spending time on this since I don't think it has been a big pain point for us. if it causes problems for us then I would start to think about changing it.

**Review by mikeharv — COMMENTED** [review]
Looks good to me! Curriculum will update lesson overviews as needed once these reach levelbuilder.

**Comment by davidsbailey** [comment]
Thanks Mike! Per our conversation, the plan is to merge this to staging and levelbuilder after Monday afternoon's deploy to production, giving you Tuesday morning to make any edits before this content reaches production.

**Comment by davidsbailey** [comment]
@mikeharv these should be live on levelbuilder now.

**Pull request merged**
